### PR TITLE
Refactored with readFileSync to read files in the order they were req…

### PR DESCRIPTION
**Pull request by mpowers1113**
…ueste

https://user-images.githubusercontent.com/88848572/143613928-7c27456e-71bb-46f2-ac94-6339029a943c.mp4

d.

### CODE REVIEW

File: node-fs-readfile/cat.js [hunk 1]
```diff
@@ -3,16 +3,14 @@ const args = process.argv;
...
-      console.log(data);
-    });
-  });
+try {
+  const input = args.slice(2);
+  for (const request of input) {
+    const eachRequest = './' + request.trim();
+    console.log(
+      fs.readFileSync(eachRequest, 'utf8')
+    );
```
Comment (thebearingedge) [inline node-fs-readfile/cat.js:12]: `readFileSync` is banned for this challenge.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Try doing this without writing to a file. You have a couple of approaches to try.

- use an array to store the file text at the appropriate index. keep a counter, when enough files have finished being read, print everything in the array.
- use recursion to read and print the files one after the other. for this approach you can't use `forEach`.

**Review by thebearingedge — APPROVED** [review]
The recursive one is right. The loop one still assumes that the last file to start reading also finished last, which isn't guaranteed.

**Comment by thebearingedge** [comment]
Actually given that you are using `let`, this is probably ok

**Comment by mpowers1113** [comment]
Phew, I did max out the call stack a few times before getting the recursion right. Fun challenge!

**Pull request merged**